modules/javafx.graphics/src/main/java/com/sun/glass/ui/headless/HeadlessWindow.java

```diff
@@ -0,0 +1,369 @@
+    protected void _exitModal(long ptr) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    protected void _requestInput(long ptr, String text, int type, double width, double height,
+            double Mxx, double Mxy, double Mxz, double Mxt,
+            double Myx, double Myy, double Myz, double Myt,
+            double Mzx, double Mzy, double Mzz, double Mzt) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    protected void _releaseInput(long ptr) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    boolean setFullscreen(boolean full) {
+        int newX = 0;
+        int newY = 0;
+        int newWidth = 0;
+        int newHeight = 0;
+        if (full) {
+            this.originalHeight = this.height;
+            this.originalWidth = this.width;
+            this.originalX = this.x;
+            this.originalY = this.y;
+            newX = 0;
+            newY = 0;
+            newWidth = screen.getWidth();
+            newHeight = screen.getHeight();
+        } else {
+            newHeight = this.originalHeight;
+            newWidth = this.originalWidth;
+            newX = this.originalX;
+            newY = this.originalY;
+        }
+        notifyResizeAndMove(newX, newY, newWidth, newHeight);
+        return full;
+    }
+
+    private void notifyResizeAndMove(int x, int y, int width, int height) {
+        HeadlessView view = (HeadlessView) getView();
+        //   if (getWidth() != width || getHeight() != height) {
+        notifyResize(WindowEvent.RESIZE, width, height);
+        if (view != null) {
+            view.notifyResize(width, height);
+        }
+        //  }
+        if (getX() != x || getY() != y) {
+            notifyMove(x, y);
+        }
+    }
+
+    public Color getColor(int lx, int ly) {
+        int mx = lx;
+        int my = ly;
+        int idx = 1000 * my + mx;
+        int rgba = frameBuffer.asIntBuffer().get(idx);
+        int a = (rgba >> 24) & 0xFF;
+        int r = (rgba >> 16) & 0xFF;
+        int g = (rgba >> 8) & 0xFF;
+        int b = rgba & 0xFF;
+
+        Color color = Color.color(
+                r / 255.0,
+                g / 255.0,
+                b / 255.0,
+                a / 255.0
+        );
+        return color;
+    }
+
+    public void getScreenCapture(int x, int y, int width, int height, int[] data, boolean scaleToFit) {
+        for (int i = 0; i < height; i++) {
+            for (int j = 0; j < width; j++) {
+                int idx = i * width + j;
+                int fidx = (y + i) * 1000 + x + j;
+                int val = frameBuffer.asIntBuffer().get(fidx);
+                data[idx] = val;
+            }
+        }
+    }
+
+    void blit(Pixels pixels) {
+        int pW = pixels.getWidth();
+        int pH = pixels.getHeight();
+        int offsetX = this.getX();
+        int offsetY = this.getY();
+        int stride = 1000;
```

Comment:
same magic 1000 here

Reply:
it's the stride used in the ByteBuffer holding the frame. I've moved it to a final int and removed all hard occurrences of 1000 .